Mathlib/Algebra/Ring/Equiv.lean

```diff
@@ -443,6 +444,61 @@ theorem piCongrRight_trans {ι : Type*} {R S T : ι → Type*}
     (piCongrRight e).trans (piCongrRight f) = piCongrRight fun i => (e i).trans (f i) :=
   rfl
 
+/-- Transport dependent functions through an equivalence of the base space.
+
+This is `Equiv.piCongrLeft'` as a `RingEquiv`. -/
+@[simps!]
+def piCongrLeft' {ι ι' : Type*} (R : ι → Type*) (e : ι ≃ ι')
+    [∀ i, NonUnitalNonAssocSemiring (R i)] :
+    ((i : ι) → R i) ≃+* ((i : ι') → R (e.symm i)) where
+  toEquiv := Equiv.piCongrLeft' R e
+  map_mul' _ _ := rfl
+  map_add' _ _ := rfl
+
+@[simp]
+theorem piCongrLeft'_symm {R : Type*} [NonUnitalNonAssocSemiring R] (e : α ≃ β) :
+    (RingEquiv.piCongrLeft' (fun _ => R) e).symm = RingEquiv.piCongrLeft' _ e.symm := by
+  simp only [piCongrLeft', RingEquiv.symm, MulEquiv.symm, Equiv.piCongrLeft'_symm]
+
+/-- Transport dependent functions through an equivalence of the base space.
+
+This is `Equiv.piCongrLeft` as a `RingEquiv`. -/
+@[simps!]
+def piCongrLeft {ι ι' : Type*} (S : ι' → Type*) (e : ι ≃ ι')
+    [∀ i, NonUnitalNonAssocSemiring (S i)] :
+    ((i : ι) → S (e i)) ≃+* ((i : ι') → S i) :=
+  (RingEquiv.piCongrLeft' S e.symm).symm
+
+/-- Splits the indices of ring `∀ (i : ι), Y i` along the predicate `p`. This is
+`Equiv.piEquivPiSubtypeProd` as a `RingEquiv`. -/
+@[simps!]
+def piEquivPiSubtypeProd {ι : Type*} (p : ι → Prop) [DecidablePred p] (Y : ι → Type*)
+    [∀ i, NonUnitalNonAssocSemiring (Y i)] :
+    ((i : ι) → Y i) ≃+* ((i : { x : ι // p x }) → Y i) × ((i : { x : ι // ¬p x }) → Y i) where
+  toEquiv := Equiv.piEquivPiSubtypeProd p Y
+  map_mul' _ _ := rfl
+  map_add' _ _ := rfl
+
+/-- Product of ring equivalences. This is `Equiv.prodCongr` as a `RingEquiv`. -/
+@[simps!]
+def prodMap {R R' S S' : Type*} [NonUnitalNonAssocSemiring R] [NonUnitalNonAssocSemiring R']
```

Comment:
Why not prodCongr?

Reply:
Yes this should be `prodCongr`! Have amended.